### DISCUSSION TITLE
Building static_h on x86_64

### DISCUSSION THREAD

**Discussion by guest271314** [facebook/hermes#1494] [General]
Hermes git revision (if applicable):
React Native version:
OS:  x86_64

## Steps To Reproduce

I have tried to build the `static_h` branch twice. Following instructions here https://medium.com/@gautam1168/compiling-typescript-to-native-code-0238d69ca582

```
mkdir hermes_workspace
cd hermes_workspace
git clone git@github.com:facebook/hermes.git
cd hermes
git checkout static_h
cd ..
```

Each occasion the system has frozen.

**Comment by tmikov** [comment]
Hi, you haven't provided any information about your system - what OS, how much RAM, which command is freezing, etc. 

Generally speaking, our build process has no power to freeze the OS, we are just running the compiler and linker in a standard way.

If you build environment is configured with too little RAM and a lot of cores, it is possible that Ninja is starting too many instances of Clang, consuming too much RAM and the system is thrashing, making it appear frozen. The solution would be to decrease the number of Ninja processes by building with `ninja -j2 shermes`, for example.

Unfortunately the Hermes team can't really provide support about the configuration of dev environment, but it is possible that members of the community can, so I am transferring this issue to discussions.
